src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -15,4 +15,423 @@
  *    limitations under the License.
  */
 
-void MatterAccessControlPluginServerInitCallback() {}
+#include <access/AccessControl.h>
+#include <access/examples/ExampleAccessControlDelegate.h>
+
+#include <app-common/zap-generated/af-structs.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/enums.h>
+
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/data-model/Encode.h>
+#include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::Access;
+
+namespace AccessControlCluster = chip::app::Clusters::AccessControl;
+
+struct AccessControlEntryCodec
+{
+    static CHIP_ERROR Convert(AuthMode from, AccessControlCluster::AuthMode & to)
+    {
+        switch (from)
+        {
+        case AuthMode::kPase:
+            to = AccessControlCluster::AuthMode::kPase;
+            break;
+        case AuthMode::kCase:
+            to = AccessControlCluster::AuthMode::kCase;
+            break;
+        case AuthMode::kGroup:
+            to = AccessControlCluster::AuthMode::kGroup;
+            break;
+        default:
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+        return CHIP_NO_ERROR;
+    }
+
+    static CHIP_ERROR Convert(AccessControlCluster::AuthMode from, AuthMode & to)
+    {
+        switch (from)
+        {
+        case AccessControlCluster::AuthMode::kPase:
+            to = AuthMode::kPase;
+            break;
+        case AccessControlCluster::AuthMode::kCase:
+            to = AuthMode::kCase;
+            break;
+        case AccessControlCluster::AuthMode::kGroup:
+            to = AuthMode::kGroup;
+            break;
+        default:
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+        return CHIP_NO_ERROR;
+    }
+
+    static CHIP_ERROR Convert(Privilege from, AccessControlCluster::Privilege & to)
```

Comment:
Likewise, it feels like the code here could be much smaller/simpler if our types were aligned better.